### PR TITLE
Images retag

### PR DESCRIPTION
**Pull request by cd1989**
An implementation of `image retag` as described in issue: https://github.com/goharbor/harbor/issues/5778

It provides an API: `POST /api/retag` to tag an image to another.

### CODE REVIEW

File: src/ui/utils/retag.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	if err != nil {
+		log.Errorf("check existence of manifest '%s:%s' error: %v", destClient.Name, destImage.Tag, err)
+		return err
+	}
+	if exist && destDigest == digest {
+		log.Infof("manifest of '%s:%s' already exist", destClient.Name, destImage.Tag)
+		return nil
+	}
+
+	if getRepoName(srcImage) != getRepoName(destImage) {
```
Comment (supereagle) [inline src/ui/utils/retag.go]: Duplicated check, can use a flag `sameRepo` to store the result.

**Comment by supereagle** [comment]
@cd1989  please fix conflict

File: docs/swagger.yaml [hunk 2]
```diff
@@ -1131,6 +1131,33 @@ paths:
...
+      responses:
+        '200':
+          description: Image retag successfully.
+        '400':
+          description: Invalid image values provided
+        '401':
+          description: User has no permission to the source project or destination project.
+        '404':
+          description: Project or repository not found.
+        '409':
```
[the remark this answers is not in the document]
Comment (cd1989) [reply docs/swagger.yaml]: `409` is returned when the desired tag is already existed.

File: src/common/utils/registry/auth/tokenauthorizer.go [hunk 3]
```diff
@@ -38,6 +38,17 @@ type tokenGenerator interface {
...
 }
 
+// UserAgentModifier adds the "User-Agent" header to the request
+type UserAgentModifier struct {
+	UserAgent string
+}
+
+// Modify adds user-agent header to the request
+func (u *UserAgentModifier) Modify(req *http.Request) error {
+	req.Header.Set(http.CanonicalHeaderKey("User-Agent"), u.UserAgent)
```
[the remark this answers is not in the document]
Comment (steven-zou) [reply src/common/utils/registry/auth/tokenauthorizer.go]: `User-Agent` is the key, `UserAgent` is a method returning the header value?

**Comment by cd1989** [comment]
> ... do you have some good ways/suggestions to improve this part?

@steven-zou Sorry, I don't have ways for the moment.

**Comment by steven-zou** [comment]
@clouderati 

Any further comments for this PR?

**Comment by steven-zou** [comment]
@cd1989 

I think it's time to merge code back to master. Let's work on the UI part of this feature now.

**Pull request merged**